File: NonCollaborative/scripts/planarsviz_checks/superseded.R
# Where the R scripts the planarsviz library replaced now live.
#
# These checks prove the library draws what the old scripts drew, by running
# the old script in memory and comparing. That makes the old scripts evidence:
# they are archived rather than deleted, and the checks stop working if they
# go. See OlderFiles/planarsviz_superseded/README.md.
#
# One place holds the path so that moving the archive again is one edit, not
# eleven. Every check sources this file and calls superseded():
#
#   source("scripts/planarsviz_checks/superseded.R")
#   src <- readLines(superseded("results", "laminar_spanchart.r"))
#
# Paths are relative to NonCollaborative/, which is where the checks run from.
#
# superseded.py is the twin of this file, for the checks written in Python.
# Both must name the same directory.

SUPERSEDED_DIR <- file.path("OlderFiles", "planarsviz_superseded")

superseded <- function(...) {
  path <- file.path(SUPERSEDED_DIR, ...)
  if (!file.exists(path)) {
    stop("Superseded script not found: ", path,
         "\nThese scripts are the evidence the porting checks compare against.",
         " If one is missing, restore it from git rather than skipping the check.",
         call. = FALSE)
  }
  path
}

File: NonCollaborative/scripts/planarsviz_checks/superseded.py
"""Where the scripts the planarsviz library replaced now live.

The Python twin of superseded.R, for the checks written in Python. Same
reasoning: these checks prove the library draws what the old scripts drew, by
comparing against what those scripts contain. That makes the old scripts
evidence -- they are archived rather than deleted, and the checks stop working
if they go. See OlderFiles/planarsviz_superseded/README.md.

One place holds the path so that moving the archive again is one edit, not
fifteen. Every check imports this and calls superseded():

    from superseded import superseded
    text = superseded("results", "laminar_four_trees.r").read_text()

Paths are resolved from NonCollaborative/, so a check works from any working
directory.
"""

from pathlib import Path

NC = Path(__file__).resolve().parents[2]
SUPERSEDED_DIR = NC / "OlderFiles" / "planarsviz_superseded"


def superseded(*parts: str) -> Path:
    """Path to one archived script, checked for existence."""
    path = SUPERSEDED_DIR.joinpath(*parts)
    if not path.exists():
        raise FileNotFoundError(
            f"Superseded script not found: {path}\n"
            "These scripts are the evidence the porting checks compare against. "
            "If one is missing, restore it from git rather than skipping the check."
        )
    return path
